doc/manual/packages.rst

```diff
@@ -309,6 +309,53 @@ Once you do this, the package manager knows your GitHub user name and can config
 You should also `upload <https://github.com/settings/ssh>`_ your public SSH key to GitHub and set up an `SSH agent <http://linux.die.net/man/1/ssh-agent>`_ on your development machine so that you can push changes with minimal hassle.
 In the future, we will make this system extensible and support other common git hosting options like `BitBucket <https://bitbucket.org>`_ and allow developers to choose their favorite.
 
+Guidelines for Naming a Package
+-------------------------------
+
+Package names should be sensible to most Julia users, *even to those who are
+not domain experts*.
+
+1. Avoid jargon. In particular, avoid acronyms unless there is minimal
+   possibility of confusion.
+
+  * It's ok to say ``USA`` if you're talking about the USA.
+
+  * It's not ok to say ``PMA``, even if you're talking about positive mental
+    attitude.
+
+2. Packages that provide most of their functionality in association with a new
+   type should have pluralized names.
+
+  * ``DataFrames`` provides the ``DataFrame`` type.
+
+  * ``BloomFilters`` provides the ``BloomFilter`` type.
+
+  * In contrast, ``JuliaParser`` provides no new type, but instead new
+    functionality in the ``JuliaParser.parse()`` function.
+
+3. Err on the side of clarity, even if clarity seems long-winded to you.
+
+  * ``RandomMatrices`` is a less ambiguous name than ``RndMat`` or ``RMT``,
+    even if the latter are shorter.
+
+4. Less systematic names may suit packages that implement one of several
+   possible approaches to its domain.
```

Comment:
Number disagreement: "to their domains".  (or "A less systematic name may suit a package that implements...").